modules/javafx.graphics/src/main/java/com/sun/scenario/animation/shared/MultiLoopClipEnvelope.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.scenario.animation.shared;
+
+import javafx.animation.Animation;
+
+/**
+ * Clip envelope for multi-cycle animations. In this case, autoReverse and cyclePosition (which can be different from ticks)
+ * are important.
+ */
+abstract class MultiLoopClipEnvelope extends ClipEnvelope {
+
+    protected boolean autoReverse;
+
+    /**
+     * The current position of the play head in its current cycle.
+     * cyclePos = ticks % cycleTicks, so 0 <= cyclePos <= cycleTicks.
+     */
+    protected long cyclePos;
+
+    protected MultiLoopClipEnvelope(Animation animation) {
+        super(animation);
+    }
+
+    protected boolean autoReverse() {
```

Comment:
That would be the usual naming convention, yes, but I find that  this can be more fluent to read. Perhaps I'm influenced by the upcoming `record`s feature. Can change.

Reply:
Yes, please change.  Animation.java also has  `isAutoReverese` method, It would be good to be uniform.